### PR TITLE
fix(bls12381): NewPublicKeyFromBytes: add missing infinity check

### PR DESCRIPTION
**Pull request by melekes**
Fixes a regression introduced in https://github.com/cometbft/cometbft/pull/3603

### CODE REVIEW

**Comment by cason** [comment]
Any idea of why this happens?

```
cometbft/crypto/bls12381 $ go test                                                                                                                                                                                    
?   	github.com/cometbft/cometbft/crypto/bls12381	[no test files]
```

**Review by cason — COMMENTED** [review]
Left some comments, but I couldn't run the tests locally.

**Comment by melekes** [comment]
> Any idea of why this happens?
> 
> ```
> cometbft/crypto/bls12381 $ go test                                                                                                                                                                                    
> ?   	github.com/cometbft/cometbft/crypto/bls12381	[no test files]
> ```

you need to add `bls12381` build flag: `go test -tags=bls12381`

File: crypto/bls12381/key_test.go [hunk 1]
```diff
@@ -167,3 +170,37 @@ func TestPubKey_MarshalJSON(t *testing.T) {
 	err = pubKey2.UnmarshalJSON(jsonBytes)
 	require.NoError(t, err)
 }
+
+func TestPubKey_NewPublicKeyFromBytes(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (melekes) [reply crypto/bls12381/key_test.go:174]: Mention how?

File: crypto/bls12381/key_test.go [hunk 2]
```diff
@@ -172,6 +172,8 @@ func TestPubKey_MarshalJSON(t *testing.T) {
 }
 
 func TestPubKey_NewPublicKeyFromBytes(t *testing.T) {
```
Comment (cason) [inline crypto/bls12381/key_test.go:174]:
```suggestion
func TestPubKey_NewPublicKeyFromInvalidBytes(t *testing.T) {
```

File: crypto/bls12381/key_test.go [hunk 3]
```diff
@@ -167,3 +170,44 @@ func TestPubKey_MarshalJSON(t *testing.T) {
...
+		pkc := pk.Serialize()
+		if pkc == nil {
+			return nil, errors.New("could not serialize pubkey")
+		}
+		return pkc, nil
+	}
+
+	invalidPubKeys := map[string]string{
+		"NotInG1":   "8123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef",
+		"InfFalseB": "800000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
```
Comment (cason) [inline crypto/bls12381/key_test.go:197]: This is a valid key, right? At least, the test passes with it even without the introduced changes.

File: crypto/bls12381/key_test.go [hunk 4]
```diff
@@ -167,3 +170,44 @@ func TestPubKey_MarshalJSON(t *testing.T) {
...
+		"InfTrueB":  "c01000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
```
Comment (cason) [inline crypto/bls12381/key_test.go:198]: The same as previous comment.

File: crypto/bls12381/key_test.go (hunk 2 above)
Comment (cason) [reply crypto/bls12381/key_test.go:174]: But now I am not sure. Are some of those keys valid?

**Review by cason — COMMENTED** [review]
Could run the tests, thank you.

File: crypto/bls12381/key_test.go (hunk 3 above)
Comment (melekes) [reply crypto/bls12381/key_test.go:197]: I copied it from the report. I think it means "infinite point A". The other one is "infinite point B". A and B are two parts. 

> One of the initially non-obvious things about BLS12-381 is that we're really dealing with two curves, not one.
https://hackmd.io/@benjaminion/bls12-381

Comment (cason) [reply crypto/bls12381/key_test.go:197]: But shouldn't it fail or not?

Comment (melekes) [reply crypto/bls12381/key_test.go]: I still think it's helpful to keep them even though they don't yield an error on `main`.

**Review by cason — APPROVED** [review]
Legit.

**Comment by cason** [comment]
But are we sure that the CI is running the added test? How does it know that it needs to add a `-tag` parameter?

**Comment by cason** [comment]
> But are we sure that the CI is running the added test? How does it know that it needs to add a -tag parameter?

Ok, it appears that it does not fail in the CI, see the last commit.

**Comment by melekes** [comment]
> > But are we sure that the CI is running the added test? How does it know that it needs to add a -tag parameter?
> 
> Ok, it appears that it does not fail in the CI, see the last commit.

Thanks for spotting it 👍 fixed in [`d527708` (#4104)](https://github.com/cometbft/cometbft/pull/4104/commits/d5277082185c3e25d003dc80317bddc8959f0466)

**Pull request merged**